dropshot/tests/fail/bad_endpoint1.rs

```diff
@@ -1,5 +1,14 @@
 // Copyright 2020 Oxide Computer Company
 
+// These "bad endpoint" tests are intended to throw compiler errors.  When a
+// compiler error is thrown, imports which haven't yet successfully been used
+// may throw "unused import" warnings. This is often a false positive, as
+// the compiler error itself blocks intended usage.
+//
+// For this test (and other bad endpoint tests) disable the import warnings
+// so output can be more precisely compared with the expected compiler error.
+#![allow(unused_imports)]
```

Comment:
It looks like previously only test 8 emitted "unused import" warnings. Did this PR cause more warnings?

I think it may be beneficial to see what actual users would see *including* unused import warnings.

Reply:
Yeah, there are a couple spots where the warnings appear, where they didn't before. I believe this is due to the difference in parsing of the first argument:

- The old version of the proc macro *forced* the first argument to be `Arc<RequestContext>`, and emitted a stream from the proc_macro implementation in more cases, relying on the compiler to catch these issues after the lexicographic substitution of the macro completed.
- The new version needs to parse that first argument, and throws errors earlier for a couple cases: an endpoint without arguments or with a non-receiver argument will cause the macro to throw an error explicitly. I added two test cases to `dropshot_endpoint/src/lib.rs` to cover these cases.

So that's a long way of saying "yes, this change does make the unused import warning appear more often, but that's mostly because it's raising the error sooner".

If you view this as a problem, I can try to force the proc macro to emit tokens, even when I know they're invalid, but that feels kinda like the wrong direction IMO.

Reply:
That makes sense and I--agreed--don't think we should emit tokens in the face of an error.

Do you think it's desirable to suppress the warnings? I think I'd rather have the tests show what a user would see.

Reply:
So, I can - but even separate from this PR, many of the `struct QueryParam` objects in the endpoint tests had the annotation `#[allow(dead_code)]`, presumably to avoid similar warnings. Should I remove those too?

Reply:
Well. I'd argue those are dead code even if the endpoint were properly formed, but your point is well-taken and I'm happy with whatever you decide is the best path here.

Reply:
I think I stand with my original stance of continuing to `#![allow]` unused imports. If I was blocking **all** warnings, I would 100% agree with you, we would want to see the output, but frankly I'd be astonished if someone saw a "compiler error + unused import warning" and prioritized the import warning.

If we can agree on that, then I believe the warning suppression will just provide a better signal/noise ratio for the test output.